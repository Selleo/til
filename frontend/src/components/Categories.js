import React from "react";
import { useSelector } from "react-redux";

const Categories = (props) => {
  const categories = useSelector((state) => state.categories);

  return categories.map((category) => (
    <p key={category.id} data-testid="category-item">
      {category.name}
    </p>
  ));
};

export default Categories;
