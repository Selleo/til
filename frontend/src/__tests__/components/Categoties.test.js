import React from "react";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import configureMockStore from "redux-mock-store";
import Categories from "../../components/Categories";

const categories = [
  {
    id: 1,
    name: "android",
  },
  {
    id: 2,
    name: "angular",
  },
  {
    id: 3,
    name: "aws",
  },
  {
    id: 4,
    name: "chrome",
  },
  {
    id: 5,
    name: "commandline",
  },
  {
    id: 6,
    name: "crystal",
  },
];

describe("<Categories/>", () => {
  it("renders properly", () => {
    const mockStore = configureMockStore([]);
    const store = mockStore({
      categories,
    });

    const component = (
      <Provider store={store}>
        <Categories />
      </Provider>
    );

    const { asFragment } = render(component);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders empty list when zero categories provided", () => {
    const mockStore = configureMockStore([]);

    const store = mockStore({
      categories: [],
    });

    const component = (
      <Provider store={store}>
        <Categories />
      </Provider>
    );

    const { queryAllByTestId } = render(component);

    const items = queryAllByTestId("category-item");

    expect(items.length).toEqual(0);
  });

  it("renders list with correct names when categories provided", () => {
    const mockStore = configureMockStore([]);

    const store = mockStore({
      categories,
    });

    const component = (
      <Provider store={store}>
        <Categories />
      </Provider>
    );

    const { queryAllByTestId } = render(component);

    const categoryNames = queryAllByTestId("category-item").map(
      (item) => item.textContent
    );

    const fakeCaregoryNames = categories.map((category) => category.name);
    expect(categoryNames).toEqual(fakeCaregoryNames);
  });
});
