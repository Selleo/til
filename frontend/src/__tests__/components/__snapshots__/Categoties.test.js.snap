// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Categories/> renders properly 1`] = `
<DocumentFragment>
  <p
    data-testid="category-item"
  >
    android
  </p>
  <p
    data-testid="category-item"
  >
    angular
  </p>
  <p
    data-testid="category-item"
  >
    aws
  </p>
  <p
    data-testid="category-item"
  >
    chrome
  </p>
  <p
    data-testid="category-item"
  >
    commandline
  </p>
  <p
    data-testid="category-item"
  >
    crystal
  </p>
</DocumentFragment>
`;
